docs/source/dataset_formats.mdx

```diff
@@ -0,0 +1,709 @@
+```
+
+### Preference
+
+A preference dataset is used for tasks where the model is trained to choose between two or more possible completions to the same prompt. This dataset includes a `"prompt"`, a `"chosen"` completion, and a `"rejected"` completion. The model is trained to select the `"chosen"` response over the `"rejected"` response.
+Some dataset may not include the `"prompt"` column, in which case the prompt is implicit and directly included in the `"chosen"` and `"rejected"` completions. We recommend using explicit prompts whenever possible.
+
+```python
+preference_example = {"prompt": "The sky is", "chosen": " blue.", "rejected": " green."}  # recommended
+# or,
+preference_example = {"chosen": "The sky is blue.", "rejected": "The sky is green."}
+```
+
+### Unpaired preference
+
+An unpaired preference dataset is similar to a preference dataset but instead of having `"chosen"` and `"rejected"` completions for the same prompt, it includes a single `"completion"` and a `"label"` indicating whether the completion is preferred or not.
+
+```python
+unpaired_preference_example = {"prompt": "The sky is", "completion": " blue.", "label": True}
+```
+
+## Which dataset format to use?
+
+Choosing the right dataset format depends on the task you are working on and the specific requirements of the TRL trainer you are using. Below is a brief overview of the dataset formats supported by each TRL trainer.
+
+| Trainer                 | Expected dataset format      |
+| ----------------------- | ---------------------------- |
+| [`BCOTrainer`]          | Unpaired preference          |
+| [`CPOTrainer`]          | Preference (explicit prompt) |
+| [`DPOTrainer`]          | Preference (explicit prompt) |
+| [`IterativeSFTTrainer`] | Unpaired preference          |
+| [`KTOTrainer`]          | Unpaired preference          |
+| [`OnlineDPOTrainer`]    | Prompt-only                  |
+| [`ORPOTrainer`]         | Preference (explicit prompt) |
+| [`PPOv2Trainer`]        | Tokenized language modeling  |
+| [`RewardTrainer`]       | Preference (implicit prompt) |
+| [`SFTTrainer`]          | Language modeling            |
+| [`XPOTrainer`]          | Prompt-only                  |
+
+<Tip>
+
+TRL trainers only support standard dataset formats. If you have a conversational dataset, you must first convert it into a standard format.
+For more information on how to work with conversational datasets, refer to the [Working with conversational datasets in TRL](#working-with-conversational-datasets-in-trl) section.
+
+</Tip>
+
+## Working with conversational datasets in TRL
+
+Conversational datasets are increasingly common, especially for training chat models. However, TRL trainers don't support conversational datasets in their raw format. These datasets must first be converted into a standard format. 
+Fortunately, TRL offers tools to easily handle this conversion, which are detailed below.
+
+### Converting a conversational dataset into a standard dataset
```

Comment:
I'm a bit confused about this mapping since I thought the `SFTTrainer` allows users to provide a dataset with a `messages` column and no preprocessing / conversion is needed. 

Overall, I think we'd like to have a situation where users can have datasets formatted in the conversational style and the chat templating step is taken care of automatically. I realise we don't have consistency in the lib right now, so I'm open to alternative ideas!

Reply:
It's annoying because `SFTTrainer` is a particular case: it supports 
- tokenized dataset
- standard dataset (ie. non-conversational)
- and conversational dataset

But it's the only trainer to support all of these.  Another reason to extend and standardise this to all trainers.

I will mention this particular case in the doc though.